micro-infra-spring-base/src/main/groovy/com/ofg/infrastructure/web/resttemplate/fluent/common/response/executor/RestExecutor.groovy

```diff
@@ -36,10 +39,17 @@ final class RestExecutor<T> {
         try {
             return exchangeInternal(params, httpMethod, responseType).get()
         } catch (ExecutionException e) {
-            throw e.cause
+            propagate(e.cause)
         }
     }
 
+    private void propagate(Throwable throwable) {
+        if (throwable instanceof TimeoutException) {
```

Comment:
Ok I can see that but why do we do a distinct wrapping of TimeoutException? Why do we need that separation between the two? Why not just do Throwables.propagate?